src/WebJobs.Extensions.DurableTask/Listener/TaskHttpActivityShim.cs

```diff
@@ -0,0 +1,101 @@
+            throw new NotImplementedException();
+        }
+
+        public async override Task<string> RunAsync(TaskContext context, string rawInput)
+        {
+            HttpRequestMessage requestMessage = await this.ReconstructHttpRequestMessage(rawInput);
+            HttpResponseMessage response = await this.httpClient.SendAsync(requestMessage);
+            DurableHttpResponse durableHttpResponse = await DurableHttpResponse.CreateDurableHttpResponseWithHttpResponseMessage(response);
+
+            return MessagePayloadDataConverter.HttpConverter.Serialize(value: durableHttpResponse, formatted: true);
+        }
+
+        private static async Task<DurableHttpResponse> CreateDurableHttpResponseAsync(HttpResponseMessage response)
+        {
+            DurableHttpResponse durableHttpResponse = new DurableHttpResponse(response.StatusCode);
+            durableHttpResponse.Headers = CreateStringValuesHeaderDictionary(response.Headers);
+            durableHttpResponse.Content = await response.Content.ReadAsStringAsync();
+            return durableHttpResponse;
+        }
+
+        internal static IDictionary<string, StringValues> CreateStringValuesHeaderDictionary(IEnumerable<KeyValuePair<string, IEnumerable<string>>> headers)
+        {
+            IDictionary<string, StringValues> newHeaders = new Dictionary<string, StringValues>();
+            if (headers != null)
+            {
+                foreach (var header in headers)
+                {
+                    newHeaders[header.Key] = new StringValues(header.Value.ToArray());
+                }
+            }
+
+            return newHeaders;
+        }
+
+        private async Task<HttpRequestMessage> ReconstructHttpRequestMessage(string serializedRequest)
+        {
+            this.serializerSettings.TypeNameHandling = TypeNameHandling.Auto;
+
+            // DeserializeObject deserializes into a List and then the first element
+            // of that list is the serialized DurableHttpRequest
+            IList<string> input = JsonConvert.DeserializeObject<List<string>>(serializedRequest, this.serializerSettings);
+            string durableHttpRequestString = input.First();
+            DurableHttpRequest durableHttpRequest = JsonConvert.DeserializeObject<DurableHttpRequest>(durableHttpRequestString, this.serializerSettings);
+
+            HttpRequestMessage requestMessage = new HttpRequestMessage(durableHttpRequest.Method, durableHttpRequest.Uri);
+            if (durableHttpRequest.Headers != null)
+            {
+                foreach (KeyValuePair<string, StringValues> entry in durableHttpRequest.Headers)
+                {
+                    requestMessage.Headers.Add(entry.Key, (IEnumerable<string>)entry.Value);
+                }
+            }
+
+            if (durableHttpRequest.Content != null)
+            {
+                requestMessage.Content = JsonConvert.DeserializeObject<HttpContent>(durableHttpRequest.Content);
```

Comment:
I think it actually needs to be this:
```suggestion
                requestMessage.Content = new StringContent(durableHttpRequest.Content);
```